khi_robot_control/CMakeLists.txt

```diff
@@ -47,10 +47,8 @@ else()
   set(krnx_lib_name krnx-x86_64)
 endif()
 
-add_library(krnx_lib SHARED IMPORTED)
-set_target_properties(krnx_lib PROPERTIES
-  IMPORTED_LOCATION ${CMAKE_CURRENT_LIST_DIR}/lib/lib${krnx_lib_name}.so)
-set(krnx_LIBRARIES krnx_lib)
+find_library(krnx_LIBRARIES lib${krnx_lib_name}.so PATHS ${CMAKE_SOURCE_DIR}/lib)
```

Comment:
Can you clarify why you use `find_library(..)` here? The `lib` directory contains these libraries, and we know they are there, so looking for them seems like it would be unnecessary.

Unless the intention is to prepare for a release of the KRNX blobs as a stand-alone system dependency, I'm not sure this change should be merged.

Reply:
I just noticed that when using an `IMPORTED` target in a `target_link_libraries(..)` statement, the full path to the library is apparently embedded in the binary/library. `find_library(..)` works differently, and does result in what @k-okada reports: only the filename of the library is embedded, and the `LD_LIBRARY_PATH` appears to be used to resolve that to an actual library on disk. After `source`-ing the `install/setup.bash` this leads to the KRNX library placed in `install/lib` being found.

So using `find_library(..)` seems like an approach to consider.